### PR TITLE
chore: split mongocli clusters commands/store

### PR DESCRIPTION
**Pull request by matt-condon**
## Proposed changes

Since existing `clusters` implementation has separate MongoCLI and AtlasCLI builders in the same file, raising this PR to decouple clusters commands

Additionally, in store/clusters, we have one function `ProjectClusters` that is currently shared between On-prem and Atlas, with the typical switch-case, so as part of this PR I created a new /store/atlas path store with only the cloud case of that function.

The other functions/endpoints can be migrated following standard "Atlas-only" procedure

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md) (if appropriate)
- [ ] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [ ] I have updated [test/README.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/README.md) (if an e2e test has been added)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: internal/store/atlas/clusters.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+//go:generate mockgen -destination=../mocks/mock_clusters.go -package=mocks github.com/mongodb/mongodb-atlas-cli/internal/store ClusterLister,AtlasClusterDescriber,OpsManagerClusterDescriber,ClusterCreator,ClusterDeleter,ClusterUpdater,AtlasClusterGetterUpdater,ClusterPauser,ClusterStarter,AtlasClusterQuickStarter,SampleDataAdder,SampleDataStatusDescriber,AtlasClusterConfigurationOptionsDescriber,AtlasSharedClusterDescriber,ClusterUpgrader,AtlasSharedClusterGetterUpgrader,AtlasClusterConfigurationOptionsUpdater,ClusterTester
+
+type ClusterLister interface {
+	ProjectClusters(string, *atlas.ListOptions) (interface{}, error)
+}
+
+// ProjectClusters encapsulate the logic to manage different cloud providers.
+func (s *Store) ProjectClusters(projectID string, opts *atlas.ListOptions) (interface{}, error) {
+	switch s.service {
```
Comment (wtrocki) [inline internal/store/atlas/clusters.go:32]: [nit] I would argue if we need switch case. All other stores in atlas do not differenciate between those as we do not need it.

**Review by wtrocki — APPROVED** [review]

File: internal/store/atlas/clusters.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+	"fmt"
+
+	"github.com/mongodb/mongodb-atlas-cli/internal/config"
+	atlas "go.mongodb.org/atlas/mongodbatlas"
+)
+
+//go:generate mockgen -destination=../mocks/mock_clusters.go -package=mocks github.com/mongodb/mongodb-atlas-cli/internal/store ClusterLister,AtlasClusterDescriber,OpsManagerClusterDescriber,ClusterCreator,ClusterDeleter,ClusterUpdater,AtlasClusterGetterUpdater,ClusterPauser,ClusterStarter,AtlasClusterQuickStarter,SampleDataAdder,SampleDataStatusDescriber,AtlasClusterConfigurationOptionsDescriber,AtlasSharedClusterDescriber,ClusterUpgrader,AtlasSharedClusterGetterUpgrader,AtlasClusterConfigurationOptionsUpdater,ClusterTester
+
+type ClusterLister interface {
+	ProjectClusters(string, *atlas.ListOptions) (interface{}, error)
```
[the remark this answers is not in the document]
Comment (matt-condon) [reply internal/store/atlas/clusters.go:27]: There's actually no upgrade as part of this PR, this is just to split a single endpoint that's currently shared between MongoCLI and AtlasCLI

File: internal/store/atlas/clusters.go (hunk 1 above)
Comment (matt-condon) [reply internal/store/atlas/clusters.go:32]: True, this store still needs to be migrated in a separate PR, so may go ahead and merge this one and suggest that change in the migration follow-up

Comment (wtrocki) [reply internal/store/atlas/clusters.go:32]: @matt-condon Feel free to leave it as we might apply some regexp to remove the switch case globally.
Sadly copilot doesn't help with this case.

**Review by wtrocki — APPROVED** [review]

File: internal/store/atlas/clusters.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (fmenezes) [reply internal/store/atlas/clusters.go:24]: it's ok, even if you change the type, no code in the command will change neither the templates or unit tests, it is just going to make the store way more useful

**Review by fmenezes — APPROVED** [review]

**Pull request merged**